### PR TITLE
1.13/go grpc

### PR DESCRIPTION
**Pull request by nfuden**
Update GRPC to v1.59.0 to address CVE: https://github.com/advisories/GHSA-m425-mq94-257g
Update xnet as well for same reasons
Bump go version while we are at it in cloud builders

### CODE REVIEW

**Comment by nfuden** [comment]
/kick dont think this is a flake any more but one last try /workspace/gloo/projects/gateway/pkg/validation/robust_client_test.go:56

File: go.mod [hunk 1]
```diff
@@ -61,16 +61,15 @@ require (
...
-	golang.org/x/mod v0.9.0
-	golang.org/x/sync v0.1.0
-	golang.org/x/tools v0.7.0
-	google.golang.org/genproto v0.0.0-20220329172620-7be39ac1afc7
-	google.golang.org/grpc v1.45.0
-	google.golang.org/protobuf v1.28.0
+	golang.org/x/mod v0.11.0
+	golang.org/x/sync v0.3.0
+	golang.org/x/tools v0.10.0
+	google.golang.org/grpc v1.55.0
```
Comment (davidjumani) [inline go.mod:71]: Shouldn't this be `1.59.0` based on the changelog or am I missing something ?

**Review by elcasteel — DISMISSED** [review]

File: go.mod (hunk 1 above)
Comment (nfuden) [reply go.mod:71]: nope! good catch!

Comment (nfuden) [reply go.mod:71]: ill update the changelog!

Comment (davidjumani) [reply go.mod:71]: Think the same goes for jhump/protoreflect

**Review by elcasteel — DISMISSED** [review]

File: go.mod [hunk 2]
```diff
@@ -34,7 +34,7 @@ require (
 	github.com/hinshun/vt10x v0.0.0-20180809195222-d55458df857c
 	github.com/imdario/mergo v0.3.12
 	github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
-	github.com/jhump/protoreflect v1.5.0
+	github.com/jhump/protoreflect v1.15.1
```
Comment (davidjumani) [inline go.mod:37]: This too needs to be updated in the changelog

**Review by elcasteel — APPROVED** [review]

**Comment by elcasteel** [comment]
I can't approve because I opened originally but LGTM!

**Comment by nfuden** [comment]
/kick Consul e2e with envoy and gloo [It] resolves consul services with hostname addresses (as opposed to IPs addresses)

**Review by davidjumani — APPROVED** [review]

**Review by elcasteel — APPROVED** [review]

**Pull request merged**